src/presentation/webapp/msf-aggregate-data/pages/MSFHomePagePresenter.ts

```diff
@@ -0,0 +1,82 @@
+import SyncRule from "../../../../models/syncRule";
+import { CompositionRoot } from "../../../CompositionRoot";
+import { D2Api } from "../../../../types/d2-api";
+import i18n from "../../../../locales";
+import { MSFSettings } from "../../../react/msf-aggregate-data/components/msf-Settings/MSFSettingsDialog";
+import { Period } from "../../../../domain/common/entities/Period";
+
+//TODO: maybe convert to class and presenter to use MVP, MVI pattern
+export async function executeAggregateData(
+    api: D2Api,
+    compositionRoot: CompositionRoot,
+    msfSettings: MSFSettings,
+    onProgressChange: (progress: string[]) => void,
+    period?: Period
+) {
+    const eventSyncRules = (
+        await SyncRule.list(api, { type: "events" }, { paging: false })
+    ).objects.slice(0, 2);
```

Comment:
During the merge I saw this ``slice``, not sure what it does but assuming existence and/or sorting of ``dataStore`` sync rules seems a bit weak to me.

Reply:
@SferaDev That's is true but It's a temporal implementation agreed with @adrianq